### PR TITLE
init refactor and clanup - async and dtos etc

### PR DESCRIPTION
**Pull request by SjoenH**

### CODE REVIEW

File: kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs [hunk 1]
```diff
@@ -1,51 +1,114 @@
...
     public List<PeriodEntity> CreatePeriods(Guid drawId, bool isSpecial, List<DrawPeriod> drawPeriods)
     {
-        List<PeriodEntity> periodEntities = new();
-        foreach (DrawPeriod drawPeriod in drawPeriods) {
-            // If the draw period is less than a week, just create one period for the whole draw period
-            // The example is Easter where there are two parts, each shorter than one week. 
-            // We expect the user to know what they are doing when they input a draw period shorter than a week
-            if (isSpecial) {
-                if (drawPeriod.Title == null) {
-                    throw new Exception("Draw period must have a title if it is shorter than a week");
```
Comment (SjoenH) [inline kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs:19]: Not checking if it is shorter than a week here... so not sure about this error

Comment (SjoenH) [reply kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs:19]: I believe there is some confusion here with regards to the special draw. 

I'm removing the special bool from a period and rather checking if the period is smaller than a week.

Comment (wigsnes) [reply kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs:19]: A normal period is a week, and then we have special periods that can be smaller, but also longer than a week not. that we are using this, But I think we have a max of two weeks in front-end, and the two weeks can only lap two weeks, so you can not have a period spanning over three different weeks at the moment. But again we do not use this currently so all special weeks for us should be less than a week.

Comment (wigsnes) [reply kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs:19]: Here it seems that if a week is a special week, you should include a title.

File: kabinizer-back-end/kabinizer-api/Services/Draw/DrawService.cs [hunk 2]
```diff
@@ -24,10 +20,86 @@ public void CreateDraw(CreateDrawDto draw)
...
+        return drawEntity;
+    }
+
+    /**
+     * Update a draw
+     */
+    public Task UpdateDraw(Guid drawId)
+    {
+        throw new NotImplementedException();
+    }
```
Comment (SjoenH) [inline kabinizer-back-end/kabinizer-api/Services/Draw/DrawService.cs:46]: @KristofferMarthinsen I'm leaving this up to you

File: kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs (hunk 1 above)
Comment (SjoenH) [reply kabinizer-back-end/kabinizer-api/Services/Period/PeriodService.cs:19]: Yes, correct.

When setting up a draw we mark it special if it has some special rules to follow. 

| Type of Draw | Eligibility |
| --- | --- |
| Normal Draw | All current employees |
| Special Draws | All current employees, except winners from the preceding year |
| Winter holiday | Families with schoolchildren |
| Autumn holiday | Families with schoolchildren  |

So we shouldn't use the `isSpecial` bool in the period to mark that it needs a title. I have implemented this change.

File: kabinizer-back-end/kabinizer-api/Services/Draw/DrawService.cs [hunk 3]
```diff
@@ -24,10 +20,86 @@ public void CreateDraw(CreateDrawDto draw)
...
+
+    /**
+     * Delete a draw
+     */
+    public async Task DeleteDraw(Guid drawId)
+    {
+        DrawEntity drawEntity = await entityContext.Draws.FindAsync(drawId) ?? throw new Exception("Draw not found");
+        entityContext.Draws.Remove(drawEntity);
+        await entityContext.SaveChangesAsync();
+    }
```
Comment (KristofferMarthinsen) [inline kabinizer-back-end/kabinizer-api/Services/Draw/DrawService.cs:56]: `   public void DeleteDraw(string id)
    {
        var draw = entityContext.Draws.Find(Guid.Parse(id)) ?? throw new Exception("Draw not found");
        entityContext.Draws.Remove(draw);
        entityContext.SaveChanges();
    }`

min kode for DeleteDraw 🥇

**Review by KristofferMarthinsen — APPROVED** [review]
lgtm

**Pull request merged**